### ISSUE TITLE
Multi select navigation state bug - to be discussed (#899)

### ISSUE THREAD

**Issue by AndyScherzinger** [owncloud/android#1412]
This is related to #899 

Scenario:
* select a single list item and click on details menu item:
![device-2016-01-07-203550](https://cloud.githubusercontent.com/assets/1315170/12180669/4f0dc84c-b57e-11e5-839e-478f6076e3d4.png)

In case you hit the back button (or click on the arrow in the action bar), the selection mode gets disabled and you are still in the details screen. Imho this is only a not very nice detail and should be resolved automatically when #1275 gets approved and merged (since it removed the details menu item). So looping in @jancborchardt and @tobiasKaminsky

**Comment by tobiasKaminsky** [comment]
Yeah, I think we can just wait for #1275.

**Comment by jancborchardt** [comment]
Yeah, probably just a non-issue as soon as the details screen is gone. The blue line I mentioned in #1434 would still be cool to fix.

**Comment by AndyScherzinger** [comment]
The blue lien isn't there ;) - The multiselection is basically a grey (kind of second action bar) hovering above the first one. So we will have to check if and how the bar can be themed/customized. If that is possible it shouldn't be to to big of a deal. But I never worked with the multiselect bar before.

**Comment by davivel** [comment]
Mmmm, I think the real problem is that the selection should not be there once the details view appears.

I mean, when the user selects one or multiple files and clicks "Remove", once the removal is finished the selection goes aways. 

With the same pattern, if the user clicks "Details", once the details view is shown, the selection should go away too. The action requested for the selection was already done.

**Comment by davivel** [comment]
Oh, I see. This happens with more commands, the action mode remains enabled after they are finished; for instance, select multiple files and download.

Don't you think would be better if the selection is cleared?

**Comment by AndyScherzinger** [comment]
Hmm, tricky. Just checked GMail and the selection persists on actions that move away from the list and come back to it but removes the selection for the delete action since this also means all selected items are gone. Looking at our action I would start start with clear the selection when calling the action.

Clearing would also be what I expect since GMail's way of not clearing it usually confuses me 0_o

**Comment by jancborchardt** [comment]
> Just checked GMail and the selection persists on actions that move away from the list and come back to it but removes the selection for the delete action since this also means all selected items are gone.

And this is the proper way to do it. Actions which move away from the list have nothing to do with the selection. They might just be to check (details for example) or a simple misclick. Actions like deletion obviously clear the selection though, since it doesn’t exist anymore.

**Comment by davivel** [comment]
> Just checked GMail and the selection persists on actions that move away from the list and come back to it 

That's fine. Gmail keeps the selection **when the user comes back**, but the selection mode is not visible in any way while the user is "away" in the other view. In this issue the problem is that the selection mode is still **visible while the user is not in the list** - see the picture. 

IMHO should be:

1. select a single item in list -> selection mode is visible
2. click "details", show details view -> selection mode is gone 
3. click back, show list view -> selection mode is visible, same single item selected

**Comment by jancborchardt** [comment]
Ah ok. Yep, of course, like you described @davivel

**Comment by davivel** [comment]
After deep review: the only transition to a different view that is problematic is details view. In every other case, the selection is remembered when going back.

The bad new is that to implement the current behaviour of rotation in tablets when details view is visible we did some tricks that make harder remembering the selection when going back. Rewriting the tricks to have a code easier to update would be awesome, but I'm sure it will need a considerable amount of time.

There are plenty of things that discourage putting more effort in that detail for the moment:
 - We already spent a lot of time in multiple selection
 - The next release is getting late.
 - The selection will be of a single file; if multiple files are selected, there is no way to click on "details"; given the file name is in the title, requesting the user to "remember" which was doesn't seem to hard.

Given this, I will grant that selection mode is disabled when the transition to details is done, and let's see if we can live with this for now.

**Comment by davivel** [comment]
This is now ready to test in https://github.com/owncloud/android/pull/1176. Any opinion is welcome :)

**Comment by mcastroSG** [comment]
Fixed

**Comment by davivel** [comment]
Great, thx :)
